dispatcher/ratelimiter.go

```diff
@@ -0,0 +1,49 @@
+package dispatcher
+
+import (
+	"context"
+	"sync"
+
+	"github.com/iotexproject/go-pkgs/cache"
+	"golang.org/x/time/rate"
+)
+
+// RateLimiter is a struct that manages a threadsafe map of rate limiters.
+type RateLimiter struct {
+	mu       sync.RWMutex
```

Comment:
we don't need this Mutex? `limiters` is created at thread-safe LRU cache

Reply:
The mutex is used to ensure that the two operations limiters.Get and limiters.Add are atomic. It is necessary